src/transformers/models/vit/image_processing_vit_fast.py

```diff
@@ -0,0 +1,349 @@
+            channels in the image. Can be overridden by the `image_mean` parameter in the `preprocess` method.
+        image_std (`float` or `List[float]`, *optional*, defaults to `IMAGENET_STANDARD_STD`):
+            Standard deviation to use if normalizing the image. This is a float or list of floats the length of the
+            number of channels in the image. Can be overridden by the `image_std` parameter in the `preprocess` method.
+    """
+
+    model_input_names = ["pixel_values"]
+    _transform_params = [
+        "do_resize",
+        "do_rescale",
+        "do_normalize",
+        "size",
+        "resample",
+        "rescale_factor",
+        "image_mean",
+        "image_std",
+        "image_type",
+    ]
+
+    def __init__(
+        self,
+        do_resize: bool = True,
+        size: Optional[Dict[str, int]] = None,
+        resample: PILImageResampling = PILImageResampling.BILINEAR,
+        do_rescale: bool = True,
+        rescale_factor: Union[int, float] = 1 / 255,
+        do_normalize: bool = True,
+        image_mean: Optional[Union[float, List[float]]] = None,
+        image_std: Optional[Union[float, List[float]]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        size = size if size is not None else {"height": 224, "width": 224}
+        size = get_size_dict(size)
+        self.do_resize = do_resize
+        self.do_rescale = do_rescale
+        self.do_normalize = do_normalize
+        self.size = size
+        self.resample = resample
+        self.rescale_factor = rescale_factor
+        self.image_mean = image_mean if image_mean is not None else IMAGENET_STANDARD_MEAN
+        self.image_std = image_std if image_std is not None else IMAGENET_STANDARD_STD
+        self._transform_settings = {}
+
+    def _build_transforms(
+        self,
+        do_resize: bool,
+        size: Dict[str, int],
+        resample: PILImageResampling,
+        do_rescale: bool,
+        rescale_factor: float,  # dummy
+        do_normalize: bool,
+        image_mean: Union[float, List[float]],
+        image_std: Union[float, List[float]],
+        image_type: ImageType,
+    ) -> "Compose":
+        """
+        Given the input settings build the image transforms using `torchvision.transforms.Compose`.
+        """
+
+        def rescale_image(image, rescale_factor):
+            return image * rescale_factor
+
+        transforms = []
+        if do_resize:
+            transforms.append(
+                Resize((size["height"], size["width"]), interpolation=pil_torch_interpolation_mapping[resample])
+            )
+
+        # Regardless of whether we rescale, all PIL and numpy values need to be converted to a torch tensor
+        # to keep cross compatibility with slow image processors
+        convert_to_tensor = image_type in (ImageType.PIL, ImageType.NUMPY)
+        if convert_to_tensor:
+            transforms.append(ToTensor())
```

Comment:
Is it worth writing a custom `ToTensor` without rescaling logic to avoid the complex rules below? We can just copy-modify the original one

Reply:
I found `PILToTensor` which thankfully does the conversion without scaling. 

I added three custom transforms regarding this comment and [your other comment](https://github.com/huggingface/transformers/pull/28847#discussion_r1606485188) about fusing the operations: 

* `NumpyToTensor`: numpy equivalent for `PILToTensor`
* `Rescale`: class which will rescale pixel values by `rescale_factor`
* `FusedRescaleNormalize`: which will do as you suggested and combine the rescale and normalize operations. Only tricky thig here is we need to make sure to convert to a floating tensor first

Reply:
Comparing with and without the fused normalize and rescale, the speed ups are between 2 - 13 %

With separate normaliza and rescale
```
List of PIL image input
Fast mean time: 0.006879540920257568 std: 0.00035099185559956325
Slow mean time: 0.010017627239227295 std: 0.00016269613600058844
1.4601 times faster
Torch tensor batch input
Fast mean time: 0.0017743961811065674 std: 6.134420692941824e-05
Slow mean time: 0.05170198488235474 std: 0.0006948744101280261
29.1670 times faster
List of PIL image input
Fast mean time: 0.002653493165969849 std: 5.281411681712124e-05
Slow mean time: 0.004650842666625976 std: 2.603507052302019e-05
1.7534 times faster
Torch tensor batch input
Fast mean time: 0.001086679458618164 std: 2.1535627834848004e-05
Slow mean time: 0.006056704044342041 std: 4.2741895905449867e-05
5.5755 times faster
```

With fused normalize and rescale:
```
List of PIL image input
Fast mean time: 0.006737879753112793 std: 0.0004794828832439867
Slow mean time: 0.009744790077209473 std: 0.0001803401154248809
1.4535 times faster
Torch tensor batch input
Fast mean time: 0.0017259061336517335 std: 0.0001853211607801645
Slow mean time: 0.050941661834716795 std: 0.0006355194041148941
29.7736 times faster
List of PIL image input
Fast mean time: 0.0025912017822265626 std: 4.1097075065458606e-05
Slow mean time: 0.005103119373321533 std: 2.3521387703663177e-05
1.9698 times faster
Torch tensor batch input
Fast mean time: 0.0009581971168518066 std: 2.3126728878460678e-05
Slow mean time: 0.006041062116622925 std: 3.5341232719903447e-05
6.3075 times faster
```

Speed ups: 
```
List of PIL image input: 1.0210245911674707 times faster
Torch tensor batch input: 1.0280954140606922 times faster
List of PIL image input: 1.024039572745956 times faster
Torch tensor batch input: 1.134087589606292 times faster
```